libs/core/src/lib/flexible-column-layout/flexible-column-layout.component.ts

```diff
@@ -0,0 +1,421 @@
+  }
+
+  /** 
+   * @hidden 
+   * Listen on window resize and update the layout
+   * call user defined custom function if provided
+   */
+  private _listenOnWindowResize(): void {
+    if (this.customOnResizeFunction) {
+        this.customOnResizeFunction();
+    } else {
+      this._subscriptions.add(
+          fromEvent(window, 'resize')
+              .pipe(debounceTime(100))
+              .subscribe(() => this._responsiveLayoutChangeHandler())
+      );
+    }
+  }
+
+  /** 
+   * @hidden 
+   * determines if the left separator should be visible
+   * and the value that will specify the direction of the arrow
+   */
+  private _getLeftColumnSeparatorValue(): ColumnSeparatorValue {
+    switch (this.layout) {
+      case ONE_COLUMN_START_FULL_SCREEN : return null;
+      case ONE_COLUMN_MID_FULL_SCREEN : return null;
+      case ONE_COLUMN_END_FULL_SCREEN : return null;
+      case TWO_COLUMNS_START_EXPANDED : return 'left';
+      case TWO_COLUMNS_MID_EXPANDED : return 'right';
+      case TWO_COLUMNS_END_EXPANDED : return null;
+      case THREE_COLUMNS_MID_EXPANDED : return 'right';
+      case THREE_COLUMNS_END_EXPANDED : return null;
+      case THREE_COLUMNS_END_MINIMIZED : return 'right';
+      case THREE_COLUMNS_START_MINIMIZED : return 'right';
+    }
+  }
+
+  /** 
+   * @hidden 
+   * determines if the right separator should be visible
+   * and the value that will specify the direction of the arrow
+   */
+  private _getRightColumnSeparatorValue(): ColumnSeparatorValue {
+    switch (this.layout) {
+      case ONE_COLUMN_START_FULL_SCREEN : return null;
+      case ONE_COLUMN_MID_FULL_SCREEN : return null;
+      case ONE_COLUMN_END_FULL_SCREEN : return null;
+      case TWO_COLUMNS_START_EXPANDED : return null;
+      case TWO_COLUMNS_MID_EXPANDED : return null;
+      case TWO_COLUMNS_END_EXPANDED : return 'right';
+      case THREE_COLUMNS_MID_EXPANDED : return 'left';
+      case THREE_COLUMNS_END_EXPANDED : return 'right';
+      case THREE_COLUMNS_END_MINIMIZED : return 'left';
+      case THREE_COLUMNS_START_MINIMIZED : return 'left';
+    }
+  }
+
+  /** 
+   * @hidden 
+   * updates the column layout based on the layout name
+   * makes a call to determine the new value of the left separator 
+   * makes a call to determine the new value of the right separator 
+   */
+  private _updateColumnLayoutParameters(): void {
+    this._columnLayout = this._layouts[this.layout];
+    this._leftColumnSeparator = this._getLeftColumnSeparatorValue();
+    this._rightColumnSeparator = this._getRightColumnSeparatorValue();
+  }
+
+  /** 
+   * @hidden 
+   * updates the layout
+   * emits an event
+   * makes a call to the helper function that will update the column layout and the separators
+   */
+  private _updateCurrentLayout(newLayout: Layout): void {
+    this.layout = newLayout;
+    this._updateColumnLayoutParameters();
+    
+    // setTimeout fixes "ExpressionChangedAfterItHasBeenCheckedError"
+    setTimeout(() => {
```

Comment:
`setTimeout` => `timer`?

Reply:
Don't understand the comment. `setTimeout` is used without a problem in the library.

Reply:
I'm about to use `rxjs.timer`

Reply:
I kept the `setTimeout` as this is the current approach of the library. Don't mind to change it to timer if everyone is ok with it.